docs(api): define and load liquid methods

jwwojak opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
Related to RLIQ-314.

This is a revision to the [labware section](https://docs.opentrons.com/v2/new_labware.html) of the API docs. The new content is in the file `new_labware.rst`. This PR adds a new section that describes the `define_liquid` and `load_liquid` methods. The new section is titled "Well Labels".

Also, as part of the revision, I've added an anchor tag to the "Commands" section of `tutorial.rst` for a cross reference link.

Sandbox demo link: http://sandbox.docs.opentrons.com/api-label-methods/v2/new_labware.html#well-labels

# Test Plan

Testing was through local builds and checked for:

- Grammar, style, spelling, element formatting (e.g. code block).
- There's a table with complicated elements in it. Those are working.
- Checked Sphinx directives and related links connecting text to methods and other docs.

Additional reviews can always help.

# Changelog

- Updates existing documentation to include description and examples for `define_liquid` and `load_liquid`.
- Adds an anchor to the "Commands" section in `tutorial.rst` that's referenced from the new content.

# Review requests

- Evaluate text for accuracy and clarity.
- Evaluate code sample. As a partial code sample (not meant to be complete), is it accurate?

# Risk assessment

Low risk. Text change to API docs.

jwwojak committed:
docs(api): define and load liquid methods

Additions to new_labware.rst that create a new section for define_liquid and load_liquid. New section is titled "Well Labels".

Also, added an anchor tag to the "Commands" section of tutorial.rst for a cross reference link.

jwwojak committed:
Merge branch 'edge' into api-label-methods

jwwojak committed:
Pre lunch push

Starting new drafts.

Saving in case the laptop falls into a lake or in the event of an alien invasion.

jwwojak committed:
Another set of revisions

A bigger push that includes:

- Revised section title
- Revised intro text, introduces methods in a 2nd graf after a method summary
- Removes large code sample in favor of 2 smaller snippets that focus on each method.
- Removes table that explained code. Explanations now live next to the code in respective H3 sections.
- Add a note to capture Ed's comments about volume amount validation in code (it doesn't validate).

More TK

jwwojak committed:
More revisions

This puts content in a better state. This push and draft are ready for additional review.

jwwojak committed:
One more final-ish push

Tweaks to the Note text

jwwojak committed:
Parallelism

Keeping text the same in the define an load liquid examples. One sentence ended "if you import it . . ." the other eneded "if you import your protocol ..."

the latter is better. Changing the former to match.

jwwojak committed:
Remove sentence from load_liquid section

Removing this text because the liquid object doesn't show up in the app if you don't use it. It's just a definition that sits there until used and then it will show up.

ecormany committed:
prose pass
emphasis on beginning of protocol
capitalization and patch version in 'Opentrons App v6.3.0'
noted that all arguments are required
moved hex explanation after snippet
added exact locations where liquid info can be found in app

ecormany committed:
code style pass
full names for first mention of methods, since they're from different classes.
ran snippets through black formatter
moved/added versionadded notices to go right after snippets

ecormany committed:
proofread pass
🇺🇸spellings of labeling, labeled
standardize capital L for liter abbrs

ecormany commented:
Committed some prose and code edits. We can take one more run through once #12269 merges and any loose ends in RQA-546 are tied up, then :shipit:

ecormany committed:
proper in-app terms

ecormany reviewed (approved):
Good to go, not expecting any other changes to method behavior! 🚢

TamarZanzouri commented on api/docs/v2/new_labware.rst:
```diff
@@ -198,6 +196,64 @@ and it will return the individual well objects in row A.
+Labeling Wells and Reservoirs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This example uses ``load_liquid`` to label the initial well location, contents, and volume (in µL) for the liquid objects created by ``define_liquid``. Notice how values of the ``liquid`` argument use the variable names ``greenWater`` and ``blueWater`` (defined above) to associate each well with a particular liquid: 
```
lets add a comment about loading 1 liquid per well? https://opentrons.atlassian.net/browse/RQA-546?focusedCommentId=76866

jwwojak replied:
Adding more text to point out that user can put more than 1 liquid in a well, but if they use these methods, the best practice is to make sure the well labels reflect this.  I'm not a scientist, but it does seem like a procedure may call for additional liquids in the same well, so the note doesn't say use one liquid only. It just (I hope) conveys the idea that you ought to avoid it. Please check to make sure this change captures or addresses the concern about multiple liquids in one well. Thanks!

jwwojak replied:
I agree. Updated text to incorporate Ed's recommendation.

jwwojak committed:
Addition to the caution note

Adding more text to point out that user can put more than 1 liquid in a well, but if they use these methods, the best practice is to make sure the well labels reflect this.  I'm not a scientist, but it does seem like a procedure may call for additional liquids in the same well, so the note doesn't say use one liquid only. It just (I hope) conveys the idea that you ought to avoid it. @ecormany maybe give this a once over to make sure I've captured this correctly. Thanks!

ecormany commented on api/docs/v2/new_labware.rst:
```diff
@@ -247,7 +247,7 @@ This example uses ``load_liquid`` to label the initial well location, contents,
 
 .. note::
-    ``load_liquid`` does not validate volume for your labware. For example, you could label a 40 µL well plate with ``volume=50``. It's your responsibility to ensure the labels you use accurately reflect the amount of liquid you can place into wells and reservoirs.
+    ``load_liquid`` does not validate volume for your labware nor does it prevent you from adding multiple liquids to each well. For example, you could label a 40 µL well plate with ``greenWater``, ``volume=50``, and then also add blue water to the well. The API won't stop you. It's your responsibility to ensure the labels you use accurately reflect the amount and types of liquid you're dispensin ...
```
"The API won't stop you." is extremely direct but in a fun way 😆 
I had deliberately used "place" rather than "dispense", since "dispensing" liquids is something the robot does in the course of the protocol, rather than something the human user does in protocol setup. Maybe `the amounts and types of liquid you plan to place into wells and reservoirs`.

jwwojak replied:
Updated with Ed's recommendation.

ecormany reviewed (approved):
Additional note covers what we want to say about multiple liquid types. + one small wording suggestion.

TamarZanzouri reviewed (approved):
Thank you for adding these changes! I think after committing Ed's suggestions we are good to go!

jwwojak committed:
Update new_labware.rst

Adding recommended changes to last sentence from @ecormany
